domain/secretbackend/service/service.go

```diff
@@ -0,0 +1,596 @@
+func (s *Service) GetSecretBackendConfigForAdmin(ctx context.Context, modelUUID coremodel.UUID) (*provider.ModelBackendConfigInfo, error) {
+	var info provider.ModelBackendConfigInfo
+	m, err := s.st.GetModel(ctx, modelUUID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	info.Configs = make(map[string]provider.ModelBackendConfig)
+	// We need to include builtin backends for secret draining and accessing those secrets while drain is in progress.
+	// TODO(secrets) - only use those in use by model
+	// For now, we'll return all backends on the controller.
+	jujuBackendID := s.controllerUUID
+	info.Configs[jujuBackendID] = provider.ModelBackendConfig{
+		ControllerUUID: s.controllerUUID,
+		ModelUUID:      m.ID.String(),
+		ModelName:      m.Name,
+		BackendConfig:  juju.BuiltInConfig(),
+	}
+	currentBackend, err := s.st.GetSecretBackend(ctx, secretbackend.BackendIdentifier{ID: m.SecretBackendID})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if currentBackend.Name == provider.Auto || currentBackend.Name == provider.Internal {
+		// TODO: The logic for determining the active backend ought to be relocated to an
+		// independent function. This function should then be invoked either when the model
+		// is initially created or when the model's secret backend configuration is reset
+		//  due to the current backend being forcefully removed. Under these circumstances,
+		// the model secret backend will never be set to `auto`; so here we can
+		// simply return the value stored in the database.
+		info.ActiveID = jujuBackendID
+	}
+
+	if m.Type == coremodel.CAAS {
+		cloud, cred, err := s.st.GetCloudCredential(ctx, modelUUID)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		k8sConfig, err := getK8sBackendConfig(cloud, cred)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		k8sBackendID := m.ID.String()
+		info.Configs[k8sBackendID] = provider.ModelBackendConfig{
+			ControllerUUID: s.controllerUUID,
+			ModelUUID:      m.ID.String(),
+			ModelName:      m.Name,
+			BackendConfig:  *k8sConfig,
+		}
+		if currentBackend.Name == provider.Auto {
+			info.ActiveID = k8sBackendID
+		}
+	}
+
+	backends, err := s.st.ListSecretBackends(ctx, true)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	for _, b := range backends {
+		if b.Name == currentBackend.Name {
+			info.ActiveID = b.ID
+		}
+		info.Configs[b.ID] = provider.ModelBackendConfig{
+			ControllerUUID: s.controllerUUID,
+			ModelUUID:      m.ID.String(),
+			ModelName:      m.Name,
+			BackendConfig: provider.BackendConfig{
+				BackendType: b.BackendType,
+				Config:      convertConfigToAny(b.Config),
+			},
+		}
+	}
+	if info.ActiveID == "" {
+		return nil, fmt.Errorf("%w: %q", secretbackenderrors.NotFound, currentBackend.Name)
+	}
+	return &info, nil
+}
+
+func convertConfigToAny(config map[string]string) map[string]interface{} {
+	if len(config) == 0 {
```

Comment:
We can just use `transform.Map` directly. W don't need the helpers or the length == 0 check.

Reply:
Given this is used in a few different places and this helper is much shorter, I prefer to have this helper func.
Also, I want to return nil instead of an empty map.